hibernate-search-plugins/org.jboss.tools.hibernate.search.runtime.common/src/org/jboss/tools/hibernate/search/runtime/common/AbstractFacadeFactory.java

```diff
@@ -0,0 +1,85 @@
+package org.jboss.tools.hibernate.search.runtime.common;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+
+import org.jboss.tools.hibernate.runtime.common.IFacade;
+import org.jboss.tools.hibernate.runtime.common.Util;
+import org.jboss.tools.hibernate.runtime.spi.ISession;
+import org.jboss.tools.hibernate.runtime.spi.ISessionFactory;
+import org.jboss.tools.hibernate.search.runtime.spi.IAnalyzer;
+import org.jboss.tools.hibernate.search.runtime.spi.IFullTextSession;
+import org.jboss.tools.hibernate.search.runtime.spi.IQueryParser;
+import org.jboss.tools.hibernate.search.runtime.spi.ISearchFactory;
+
+public abstract class AbstractFacadeFactory implements IFacadeFactory {
```

Comment:
I agree with you here. But i think it's a question of the whole architecture. Because at the moment we do really have lot's of reflections for common operations in hibernate tools, e.g.: https://github.com/jbosstools/jbosstools-hibernate/blob/3cdc4412072b102c5cd1db5799eeb6a177aa5bb1/plugins/org.jboss.tools.hibernate.runtime.common/src/org/jboss/tools/hibernate/runtime/common/AbstractJDBCReaderFacade.java and other `AbstractFacade`s